src/JustEat.StatsD/IpTransport.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net.Sockets;
+using JustEat.StatsD.EndpointLookups;
+
+namespace JustEat.StatsD
+{
+    public class IpTransport : IStatsDTransport
+    {
+        private readonly IPEndPointSource _endpointSource;
+
+        public IpTransport(IPEndPointSource endPointSource)
+        {
+            if (endPointSource == null)
+            {
+                throw new ArgumentNullException(nameof(endPointSource));
+            }
+            _endpointSource = endPointSource;
```

Comment:
I suggest that a review of using new syntax throughout be left to a later PR. But let's actually do it some time soon.